core/primitives/src/runtime/config_store.rs

```diff
@@ -99,6 +99,22 @@ impl RuntimeConfigStore {
         Self { store }
     }
 
+    /// Create store of runtime configs for the given chain id.
+    ///
+    /// For mainnet and other chains except testnet we don't need to override runtime config for
+    /// first protocol versions.
+    /// For testnet, runtime config for genesis block was (incorrectly) different, that's why we
+    /// need to override it specifically to preserve compatibility.
+    pub fn for_chain_id(chain_id: &str) -> Self {
```

Comment:
I prefer to keep even chain-related entities outside runtime-related ones because their dependency is the other way round.
Can we move it to `core/primitives` somewhere instead?

Reply:
Although that said, it feels like a way more natural place to have the constructor for config store within config store. This one specifically uses a string instead of any components from chain.

I fear going forward it might be very hard to locate this function in case anyone else wants to use it which would lead to code duplication.

Reply:
Keeping the function in config_store for now. Let me know if you feel strongly and I'll update in a separate PR!